pkg/build/builder/dockerutil.go

```diff
@@ -62,15 +81,27 @@ func pushImage(client DockerClient, name string, authConfig docker.AuthConfigura
 	if glog.V(5) {
 		opts.OutputStream = os.Stderr
 	}
-	var err error
+
 	var retriableError = false
 
+	eventWatch, err := watchForFailedPushEvent(eventClient, ref.Namespace, ref.Name)
```

Comment:
And upstream is wrong - we should either carry a patch or fix the issue
there.  Builds should not be coupled to events through to a registry.

On Thu, Apr 7, 2016 at 11:51 AM, Clayton Coleman <ccoleman@redhat.com>
wrote:

> I don't want the build client ever knowing about this.  I'll accept that
> the user experience is worse because upstream is blocking a better
> message.  Builds should *never*  be making assumptions about anything on
> the server side.  Please remove this.
>
> On Thu, Apr 7, 2016 at 11:45 AM, Michal Minar <notifications@github.com>
> wrote:
>
>> In pkg/build/builder/dockerutil.go
>> <https://github.com/openshift/origin/pull/8195#discussion_r58894761>:
>>
>> >  	var retriableError = false
>> >
>> > +	eventWatch, err := watchForFailedPushEvent(eventClient, ref.Namespace, ref.Name)
>>
>> The events related to ImageStreams should be pretty low on frequency. I
>> don't think this is an ideal solution but I consider it safe given the
>> filtering used. If, for some reason, we fail to catch an event, the user
>> will still see the original requested access to the resource is denied
>> message.
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly or view it on GitHub
>> <https://github.com/openshift/origin/pull/8195/files/182833487b48f39d14835bb857f60d2f9aad8099#r58894761>
>>
>
>